src/gurt/telemetry.c

```diff
@@ -1295,6 +1329,88 @@ d_tm_print_stats(FILE *stream, struct d_tm_stats_t *stats, int format)
 	fprintf(stream, ", samples: %lu]", stats->sample_size);
 }
 
+static int
+_d_tm_reset_node(struct d_tm_context *ctx, struct d_tm_node_t *node)
+{
+	struct d_tm_metric_t	*metric_data = NULL;
+	struct d_tm_stats_t	*dtm_stats = NULL;
+	struct d_tm_histogram_t *dtm_histogram = NULL;
+	struct d_tm_shmem_hdr	*shmem = NULL;
+	int			 rc;
+
+	if (ctx == NULL || node == NULL)
+		return -DER_INVAL;
+
+	rc = validate_node_ptr(ctx, node, &shmem);
+	if (rc != 0)
+		return rc;
+
+	metric_data = conv_ptr(shmem, node->dtn_metric);
+	if (metric_data == NULL)
+		return -DER_METRIC_NOT_FOUND;
+
+	dtm_stats = conv_ptr(shmem, metric_data->dtm_stats);
+	dtm_histogram = conv_ptr(shmem, metric_data->dtm_histogram);
+	d_tm_node_lock(node);
+	memset(&metric_data->dtm_data, 0, sizeof(metric_data->dtm_data));
+	if (dtm_stats != NULL)
+		memset(dtm_stats, 0, sizeof(*dtm_stats));
+
+	if (dtm_histogram != NULL) {
+		int i;
+
+		for (i = 0; i < dtm_histogram->dth_num_buckets; i++) {
+			struct d_tm_node_t	*bucket;
+
+			bucket = dtm_histogram->dth_buckets[i].dtb_bucket;
+			_d_tm_reset_node(ctx, bucket);
+		}
+	}
+
+	d_tm_node_unlock(node);
+	return DER_SUCCESS;
+}
+
+static void
+d_tm_reset_node(struct d_tm_context *ctx, struct d_tm_node_t *node, int level,
+		char *path, int format, int opt_fields, FILE *stream)
```

Comment:
Since this library is an API, maybe this function could be public. I can imagine a scenario where someone may want to reset a specific metric only.

Reply:
oh, yes, reset can share the same parameter with print.  Though I choose to export d_tm_iterate(), see above comments.